File: Frontend/src/app/Model/ProductRuleJob.ts
import { BusinessUnitProcessDto } from "./BusinessUnit";
import { JobProcessedDetails } from "./ProcessDetails";

export interface ProductRuleJob {
  id: number | undefined | null;
  productName: string | undefined | null;
  sizeCategory: string | undefined | null;
  size: string | undefined | null;
  category: string | undefined | null;
  processList?: BusinessUnitProcessDto[];
  processedDetailList: JobProcessedDetails[];
  type: string | undefined | null;
}

File: src/main/java/com/PrintLab/model/ProductRuleJob.java
package com.PrintLab.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class ProductRuleJob {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String productName;
    private String sizeCategory;
    @Column(columnDefinition = "TEXT")
    private String size;
    private String category;
    @ManyToMany
    @JoinTable(
            name = "job_process",
            joinColumns = @JoinColumn(name = "product_rule_job_id"),
            inverseJoinColumns = @JoinColumn(name = "business_unit_process_id")
    )
    private List<BusinessUnitProcess> processList;
    @OneToMany(mappedBy = "productRuleJob", cascade = CascadeType.ALL)
    private List<JobProcessedDetails> processedDetailList;
    private String type;
}